File: 88-merge-sorted-array/88-merge-sorted-array.java
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        int avilabeIndex = nums1.length-1;
        m--;
        n--;
        while(m>=0 && n>=0){
            if(nums1[m]>=nums2[n]){
                nums1[avilabeIndex]=nums1[m];
                m--;
            }else{
                nums1[avilabeIndex] = nums2[n];
                n--;
            }
            avilabeIndex--;
        }
        while(n>=0){
            nums1[avilabeIndex] = nums2[n];
            n--;
            avilabeIndex--;
        }
       
    }
}

File: 88-merge-sorted-array/88-merge-sorted-array.js
/**
 * @param {number[]} nums1
 * @param {number} m
 * @param {number[]} nums2
 * @param {number} n
 * @return {void} Do not return anything, modify nums1 in-place instead.
 */
var merge = function(nums1, m, nums2, n) {
    let i = m-1;
    let j = n-1;
    let k = nums1.length-1;
   while(i>=0 && j>=0){
        if(nums2[j]>=nums1[i]){
            nums1[k] = nums2[j];
            j--;
        }else{
            nums1[k] = nums1[i];
        i--;
        }
        
        k--;
    }
    while(j>=0){
        nums1[k] = nums2[j];
        j--;
        k--;
    }
    
};